File: pw_capture.py
from playwright.sync_api import sync_playwright, TimeoutError as PWTimeout
import sys

SCREENSHOT_PATH = r"c:\Pwc_Learning\ClaudePlayground\InvestmentAdvisoryAIAgent-main\InvestmentAdvisoryAIAgent-main\screenshots\01_advisory_home.png"

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    context = browser.new_context(viewport={"width": 1280, "height": 800})
    page = context.new_page()

    print("Step 1: Navigating to http://localhost:8501 ...")
    try:
        page.goto("http://localhost:8501", timeout=30000)
        print("  -> Navigation complete")
    except Exception as e:
        print(f"  -> Navigation error: {e}")

    print("Step 2: Waiting for 'Investment Advisory AI Agent' text (timeout=15000ms)...")
    try:
        page.wait_for_selector("text=Investment Advisory AI Agent", timeout=15000)
        print("  -> FOUND: Title text 'Investment Advisory AI Agent'")
    except PWTimeout:
        print("  -> WARN: Title text not found within 15s, proceeding anyway")
    except Exception as e:
        print(f"  -> ERROR waiting for selector: {e}")

    # Extra wait for Streamlit to fully render
    page.wait_for_timeout(2000)

    print("Step 3: Taking full-page screenshot...")
    try:
        page.screenshot(path=SCREENSHOT_PATH, full_page=True)
        print(f"  -> Screenshot saved: {SCREENSHOT_PATH}")
    except Exception as e:
        print(f"  -> Screenshot error: {e}")

    print("Step 4: Extracting all visible text...")
    try:
        # Get full inner text of the body
        body_text = page.evaluate("""() => {
            const allNodes = [];
            const walker = document.createTreeWalker(
                document.body,
                NodeFilter.SHOW_TEXT,
                {
                    acceptNode: (node) => {
                        const parent = node.parentElement;
                        if (!parent) return NodeFilter.FILTER_REJECT;
                        const style = window.getComputedStyle(parent);
                        if (style.display === 'none' || style.visibility === 'hidden' || style.opacity === '0')
                            return NodeFilter.FILTER_REJECT;
                        const text = node.textContent.trim();
                        if (!text) return NodeFilter.FILTER_REJECT;
                        return NodeFilter.FILTER_ACCEPT;
                    }
                }
            );
            let node;
            while ((node = walker.nextNode())) {
                const t = node.textContent.trim();
                if (t) allNodes.push(t);
            }
            return allNodes;
        }""")

        print("\n=== ALL VISIBLE TEXT ON PAGE ===")
        for line in body_text:
            print(f"  {line}")
    except Exception as e:
        print(f"  -> Error extracting text: {e}")

    print("\nStep 5: Extracting UI elements...")
    try:
        elements = page.evaluate("""() => {
            const result = {buttons: [], inputs: [], dropdowns: [], links: [], headings: [], labels: [], alerts: []};

            // Buttons
            document.querySelectorAll('button, [role="button"], .stButton button').forEach(b => {
                const t = b.textContent.trim();
                if (t) result.buttons.push(t);
            });

            // Inputs
            document.querySelectorAll('input, textarea').forEach(inp => {
                result.inputs.push({
                    type: inp.type,
                    placeholder: inp.placeholder || '',
                    id: inp.id || '',
                    name: inp.name || '',
                    value: inp.value || ''
                });
            });

            // Selects
            document.querySelectorAll('select').forEach(s => {
                const opts = Array.from(s.options).map(o => o.text);
                result.dropdowns.push({id: s.id || '', options: opts});
            });

            // Streamlit selectbox (custom dropdowns)
            document.querySelectorAll('[data-testid="stSelectbox"]').forEach(sb => {
                result.dropdowns.push('Streamlit Selectbox: ' + sb.textContent.trim().substring(0, 100));
            });

            // Headings
            document.querySelectorAll('h1, h2, h3, h4, h5, h6').forEach(h => {
                result.headings.push(h.tagName + ': ' + h.textContent.trim());
            });

            // Streamlit alerts/errors
            document.querySelectorAll('.stAlert, .stException, [data-testid="stNotificationContentError"], [data-testid="stNotificationContentWarning"], [data-testid="stNotificationContentInfo"]').forEach(a => {
                result.alerts.push(a.textContent.trim().substring(0, 200));
            });

            return result;
        }""")

        print("\n=== UI ELEMENTS ===")
        print(f"\nHEADINGS ({len(elements['headings'])}):")
        for h in elements['headings']:
            print(f"  {h}")

        print(f"\nBUTTONS ({len(elements['buttons'])}):")
        for b in elements['buttons']:
            print(f"  [{b}]")

        print(f"\nINPUTS ({len(elements['inputs'])}):")
        for i in elements['inputs']:
            print(f"  type={i['type']} placeholder=\"{i['placeholder']}\" id=\"{i['id']}\"")

        print(f"\nDROPDOWNS/SELECTBOXES ({len(elements['dropdowns'])}):")
        for d in elements['dropdowns']:
            if isinstance(d, dict):
                print(f"  id={d['id']} options={d['options']}")
            else:
                print(f"  {d}")

        if elements['alerts']:
            print(f"\nALERTS/ERRORS ({len(elements['alerts'])}):")
            for a in elements['alerts']:
                print(f"  ALERT: {a}")
        else:
            print("\nNO ALERTS OR ERRORS DETECTED")

    except Exception as e:
        print(f"  -> Error extracting elements: {e}")

    # Get page title
    print(f"\n=== PAGE TITLE: {page.title()} ===")
    print(f"=== PAGE URL: {page.url} ===")

    browser.close()
    print("\nDone.")

File: pw_capture.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({ viewport: { width: 1280, height: 800 } });
  const page = await context.newPage();

  try {
    console.log('Navigating to http://localhost:8501 ...');
    await page.goto('http://localhost:8501', { timeout: 30000 });

    // Wait for the title text
    try {
      await page.waitForSelector('text=Investment Advisory AI Agent', { timeout: 15000 });
      console.log('FOUND: Investment Advisory AI Agent text');
    } catch (e) {
      console.log('WARN: Could not find title text within 15s, proceeding anyway...');
    }

    // Take full-page screenshot
    const screenshotPath = 'c:\Pwc_Learning\ClaudePlayground\InvestmentAdvisoryAIAgent-main\InvestmentAdvisoryAIAgent-main\screenshots\01_advisory_home.png';
    await page.screenshot({ path: screenshotPath, fullPage: true });
    console.log('Screenshot saved to: ' + screenshotPath);

    // Get all visible text
    const bodyText = await page.evaluate(() => {
      // Get all text nodes visible on the page
      const allText = [];
      
      // Get page title
      allText.push('=== PAGE TITLE: ' + document.title + ' ===');
      
      // Get all text content
      const walker = document.createTreeWalker(
        document.body,
        NodeFilter.SHOW_TEXT,
        {
          acceptNode: (node) => {
            const parent = node.parentElement;
            if (!parent) return NodeFilter.FILTER_REJECT;
            const style = window.getComputedStyle(parent);
            if (style.display === 'none' || style.visibility === 'hidden') return NodeFilter.FILTER_REJECT;
            const text = node.textContent.trim();
            if (!text) return NodeFilter.FILTER_REJECT;
            return NodeFilter.FILTER_ACCEPT;
          }
        }
      );
      
      let node;
      while ((node = walker.nextNode())) {
        const text = node.textContent.trim();
        if (text && text.length > 0) {
          allText.push(text);
        }
      }
      return allText.join('\n');
    });

    console.log('\n=== ALL VISIBLE TEXT ON PAGE ===');
    console.log(bodyText);

    // Get interactive elements
    const elements = await page.evaluate(() => {
      const result = [];
      
      // Buttons
      const buttons = document.querySelectorAll('button, [role="button"]');
      buttons.forEach(b => {
        const text = b.textContent.trim();
        if (text) result.push('BUTTON: ' + text);
      });
      
      // Inputs
      const inputs = document.querySelectorAll('input, textarea');
      inputs.forEach(inp => {
        const label = inp.placeholder || inp.name || inp.id || inp.type || 'unknown';
        result.push('INPUT: type=' + inp.type + ' placeholder="' + (inp.placeholder || '') + '" id="' + (inp.id || '') + '"');
      });
      
      // Selects / dropdowns
      const selects = document.querySelectorAll('select');
      selects.forEach(s => {
        const opts = Array.from(s.options).map(o => o.text).join(', ');
        result.push('DROPDOWN: id=' + (s.id || 'unknown') + ' options=[' + opts + ']');
      });
      
      // Links
      const links = document.querySelectorAll('a');
      links.forEach(a => {
        const text = a.textContent.trim();
        if (text) result.push('LINK: ' + text + ' href=' + a.href);
      });
      
      return result.join('\n');
    });

    console.log('\n=== UI ELEMENTS ===');
    console.log(elements);

    // Check for any error messages
    const errors = await page.evaluate(() => {
      const errTexts = [];
      const errorSelectors = ['.stException', '.stError', '[data-testid="stException"]', '.element-container .stAlert'];
      errorSelectors.forEach(sel => {
        const els = document.querySelectorAll(sel);
        els.forEach(el => errTexts.push('ERROR/ALERT: ' + el.textContent.trim()));
      });
      return errTexts.join('\n');
    });

    if (errors) {
      console.log('\n=== ERRORS/ALERTS ===');
      console.log(errors);
    } else {
      console.log('\n=== NO ERRORS DETECTED ===');
    }

  } catch (err) {
    console.error('FATAL ERROR: ' + err.message);
    // Still try to screenshot on error
    try {
      await page.screenshot({ path: 'c:\Pwc_Learning\ClaudePlayground\InvestmentAdvisoryAIAgent-main\InvestmentAdvisoryAIAgent-main\screenshots\01_advisory_home.png', fullPage: true });
      console.log('Error screenshot saved.');
    } catch (e2) {}
  }

  await browser.close();
})();
